budgie-brightness-controller/src/helpers/DimHelper.vala

```diff
@@ -161,5 +174,11 @@ public class DimHelper
         });
         configHelper.Write(data);
     }
+
+    public bool NightlightOn() {
+        // Returns true if Nightlight mode is currently on
+        return color_settings.nightlight_active;

```

Comment:
If it reaches here ... but color_settings is still null then possible crash scenario